controllers/extension-certificate-service/pkg/controller/certservice/actuator.go

```diff
@@ -0,0 +1,335 @@
+
+	"github.com/gardener/gardener-extensions/controllers/extension-certificate-service/pkg/controller/certservice/internal"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apiserver/pkg/authentication/user"
+
+	"github.com/gardener/gardener/pkg/utils/secrets"
+
+	"github.com/gardener/gardener-extensions/controllers/extension-certificate-service/pkg/imagevector"
+
+	"github.com/gardener/gardener/pkg/chartrenderer"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	"github.com/gardener/gardener-extensions/pkg/controller"
+
+	"github.com/gardener/gardener-extensions/controllers/extension-certificate-service/pkg/utils"
+	"github.com/gardener/gardener-extensions/pkg/util"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+
+	"github.com/go-logr/logr"
+
+	configv1alpha1 "github.com/gardener/gardener-extensions/controllers/extension-certificate-service/pkg/apis/config/v1alpha1"
+	"github.com/gardener/gardener-extensions/pkg/controller/extension"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ActuatorName is the name of the Certificate Service actuator.
+const ActuatorName = "certificate-service-actuator"
+
+// NewActuator returns an actuator responsible for Extension resources.
+func NewActuator(config configv1alpha1.CertificateServiceConfiguration) extension.Actuator {
+	return &actuator{
+		logger:            log.Log.WithName(ActuatorName),
+		certServiceConfig: config,
+	}
+}
+
+type actuator struct {
+	client client.Client
+	config *rest.Config
+
+	certServiceConfig configv1alpha1.CertificateServiceConfiguration
+
+	logger logr.Logger
+}
+
+// CreateOrUpdate the Extension resource.
+func (a *actuator) CreateOrUpdate(ctx context.Context, ex *extensionsv1alpha1.Extension) error {
+	namespace := ex.GetNamespace()
+	cluster, err := controller.GetCluster(ctx, a.client, namespace)
+	if err != nil {
+		return err
+	}
+
+	if err := a.createCertBroker(ctx, cluster.Shoot, namespace); err != nil {
+		return err
+	}
+
+	secret, err := a.createKubeconfigForRBAC(ctx, namespace)
+	if err != nil {
+		return fmt.Errorf("Error getting Kubeconfig to deploy RBAC manifests: %v", err)
+	}
+
+	kubecfg, err := clientcmd.RESTConfigFromKubeConfig(secret.Data["kubeconfig"])
+	if err != nil {
+		return fmt.Errorf("Error creating Kubeconfig to deploy RBAC manifests: %v", err)
+	}
+	return a.createRBAC(ctx, kubecfg)
+}
+
+// Delete the Extension resource.
+func (a *actuator) Delete(ctx context.Context, ex *extensionsv1alpha1.Extension) error {
+	namespace := ex.GetNamespace()
+	if err := a.deleteCertBroker(ctx, namespace); err != nil {
+		return err
+	}
+
+	secret, err := a.createKubeconfigForRBAC(ctx, namespace)
```

Comment:
For the deletion you only need the name/namespace, so maybe add this information as a `const` variable somewhere that you can access here to delete the secret. Otherwise you would unnecessarily generate a certificate each and every time.

Reply:
Since we can use the `gardener` secret, we can skip the deletion anyway.